test/Integration/IntegrationTestBase.cs

```diff
@@ -164,6 +167,72 @@ private void ExecuteAllScriptsInFolder(string folder)
             }
         }
 
+        /// <summary>
+        /// Build the samples-java and test-java projects.
+        /// </summary>
+        private void BuildJavaFunctionApps()
+        {
+            // To build Java function apps while running tests locally, you need to have Maven installed
+            // and the MAVEN_PATH environment variable set to the path to the mvn.cmd file.
+            // The Java projects are built using the Maven task in ADO pipelines.
+            string mavenPath = Environment.GetEnvironmentVariable("MAVEN_PATH");
+            if (!string.IsNullOrEmpty(mavenPath))
+            {
+                // Build samples-java
+                string samplesJavaPath = Path.Combine(GetPathToBin(), "SqlExtensionSamples", "Java");
+                this.BuildJavaFunctionApp(mavenPath, samplesJavaPath);
+
+                // Build test-java
+                string testJavaPath = Path.Combine(GetPathToBin(), "..", "..", "..", "Integration", "test-java");
+                this.BuildJavaFunctionApp(mavenPath, testJavaPath);
+            }
+        }
+
+        /// <summary>
+        /// Run `mvn clean package` to build the Java function app.
+        /// </summary>
+        private void BuildJavaFunctionApp(string mavenPath, string workingDirectory)
+        {
+            this.Mvn = new Process()
+            {
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = mavenPath,
+                    Arguments = "clean package",
+                    WorkingDirectory = workingDirectory,
+                    WindowStyle = ProcessWindowStyle.Hidden,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    UseShellExecute = false
+                }
+            };
+
+            var taskCompletionSource = new TaskCompletionSource<bool>();
+            void SignalStartupHandler(object sender, DataReceivedEventArgs e)
+            {
+                if (e.Data?.Contains("BUILD SUCCESS") == true)
+                {
+                    taskCompletionSource.SetResult(true);
+                }
+            };
+            this.Mvn.OutputDataReceived += SignalStartupHandler;
+
+            this.Mvn.Start();
+            this.Mvn.OutputDataReceived += this.GetTestOutputHandler();
+            this.Mvn.ErrorDataReceived += this.GetTestOutputHandler();
+            this.Mvn.BeginOutputReadLine();
+            this.Mvn.BeginErrorReadLine();
+
+            this.LogOutput($"Building Java function app...");
+
+            const int buildJavaAppTimeoutInSeconds = 60;
+            bool isCompleted = taskCompletionSource.Task.Wait(TimeSpan.FromSeconds(buildJavaAppTimeoutInSeconds));
```

Comment:
Instead of waiting for a build message couldn't you just wait for the process to exit and then check the exit code? That'd be easier/safer in case the message changes for some reason (and also guarantees that the process will have actually exited).

You can still have a timeout on the wait to ensure we aren't waiting for a ridiculously long time.

This also would mean you don't have to redirect the stdout which means you can use shell execute and get rid of the requirement to have MAVEN_PATH defined.